src/stan/io/array_var_context.hpp

```diff
@@ -202,17 +226,14 @@ class array_var_context : public var_context {
    *
    * @param name Name of variable.
    * @return Values of variable.
+   *
    */
   std::vector<double> vals_r(const std::string& name) const {
     if (contains_r_only(name)) {
       return (vars_r_.find(name)->second).first;
     } else if (contains_i(name)) {
       std::vector<int> vec_int = (vars_i_.find(name)->second).first;
-      std::vector<double> vec_r(vec_int.size());
-      for (size_t ii = 0; ii < vec_int.size(); ii++) {
-        vec_r[ii] = vec_int[ii];
-      }
-      return vec_r;
+      return {vec_int.begin(), vec_int.end()};
```

Comment:
These are so nice!

Reply:
Ty! For the `var` construction it looks a little weirder but for these constructors it's pretty nice! I'm fine with being a little loosey-goosey on when we use the initializer list return style. Though the math lib has a script now that will auto-fix the vars to initializer lists if we ever want it